samza-azure/src/main/java/org/apache/samza/checkpoint/azure/AzureCheckpointManager.java

```diff
@@ -0,0 +1,139 @@
+import org.apache.samza.checkpoint.Checkpoint;
+import org.apache.samza.checkpoint.CheckpointManager;
+import org.apache.samza.config.AzureConfig;
+import org.apache.samza.container.TaskName;
+import org.apache.samza.system.SystemStreamPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class AzureCheckpointManager implements CheckpointManager {
+  private static final Logger LOG = LoggerFactory.getLogger(AzureCheckpointManager.class.getName());
+  private Set<TaskName> taskNames = new HashSet<>();
+  private static final String PARTITION_KEY = "PartitionKey";
+
+  public static final String CHECKPOINT_MANAGER_TABLE_NAME = "Task_Checkpoints";
+  // Define the connection-string with your values.
+  private String storageConnectionString =
+          "DefaultEndpointsProtocol=http;" +
+                  "AccountName=your_storage_account;" +
+                  "AccountKey=your_storage_account_key";
+  private AzureClient azureClient;
+  private CloudTable cloudTable;
+
+  AzureCheckpointManager(AzureConfig azureConfig) {
+    storageConnectionString = azureConfig.getAzureConnect();
+    azureClient = new AzureClient(storageConnectionString);
+  }
+
+  @Override
+  public void start() {
+    try {
+      // Create the table if it doesn't exist.
+      cloudTable = azureClient.getTableClient().getTableReference(CHECKPOINT_MANAGER_TABLE_NAME);
+      cloudTable.createIfNotExists();
+    } catch (URISyntaxException e) {
+      LOG.error("Connection string {} specifies an invalid URI while creating checkpoint table.",
+              storageConnectionString);
+    } catch (StorageException e) {
+      LOG.error("Azure Storage failed when creating a table", e);
+    }
+  }
+
+  @Override
+  public void register(TaskName taskName) {
+    taskNames.add(taskName);
+  }
+
+  @Override
+  public void writeCheckpoint(TaskName taskName, Checkpoint checkpoint) {
+    if (!taskNames.contains(taskName)) {
+      throw new SamzaException("writing checkpoint of unregistered task");
+    }
+
+    TableBatchOperation batchOperation = new TableBatchOperation();
+
+    checkpoint.getOffsets().forEach((ssp, offset) -> {
+        TaskCheckpointEntity taskCheckpoint = new TaskCheckpointEntity(taskName.toString(), ssp.getSystem());
+        taskCheckpoint.setStreamName(ssp.getStream());
+        taskCheckpoint.setPartitionId(ssp.getPartition().getPartitionId());
+        taskCheckpoint.setOffset(offset);
+        batchOperation.insertOrReplace(taskCheckpoint);
+      });
+
+    try {
+      cloudTable.execute(batchOperation);
+    } catch (StorageException e) {
+      LOG.error("Connection string {} specifies an invalid key.", storageConnectionString);
+      throw new AzureException(e);
+    }
+  }
+
+  @Override
+  public Checkpoint readLastCheckpoint(TaskName taskName) {
```

Comment:
Would be nice to add a diagram with the fields in the relational table, and how checkpoints corresponding to each of the tasks are stored and retrieved.